experimental/ragas_experimental/backends/notion.py

```diff
@@ -0,0 +1,376 @@
+                    # Notion rich text has a limit, truncate if needed
+                    if len(content) > 2000:
+                        content = content[:1997] + "..."
+                        logger.warning(f"Truncated {key} content to fit Notion limits")
+                    notion_props[key] = {"rich_text": [{"text": {"content": content}}]}
+                elif key in ["Created_At", "Updated_At"]:
+                    if isinstance(value, str):
+                        notion_props[key] = {"date": {"start": value}}
+                    elif isinstance(value, datetime):
+                        notion_props[key] = {"date": {"start": value.isoformat()}}
+            else:
+                # Convert other properties to rich text as JSON
+                content = json.dumps(value) if not isinstance(value, str) else value
+                if len(content) > 2000:
+                    content = content[:1997] + "..."
+                    logger.warning(f"Truncated {key} content to fit Notion limits")
+                notion_props[key] = {"rich_text": [{"text": {"content": content}}]}
+
+        return notion_props
+
+    def _convert_from_notion_properties(
+        self, page: t.Dict[str, t.Any]
+    ) -> t.Dict[str, t.Any]:
+        """Convert Notion page properties to Python data."""
+        properties = page.get("properties", {})
+        result = {}
+
+        for prop_name, prop_data in properties.items():
+            # Handle both actual Notion API format and test format
+            prop_type = prop_data.get("type")
+
+            # If no type field, infer from the data structure (for test compatibility)
+            if prop_type is None:
+                if "title" in prop_data:
+                    prop_type = "title"
+                elif "select" in prop_data:
+                    prop_type = "select"
+                elif "rich_text" in prop_data:
+                    prop_type = "rich_text"
+                elif "date" in prop_data:
+                    prop_type = "date"
+
+            if prop_type == "title":
+                title_content = prop_data.get("title", [])
+                result[prop_name] = "".join(
+                    [t.get("plain_text", "") for t in title_content]
+                )
+            elif prop_type == "select":
+                select_data = prop_data.get("select")
+                result[prop_name] = select_data.get("name") if select_data else None
+            elif prop_type == "rich_text":
+                rich_text_content = prop_data.get("rich_text", [])
+                text = "".join([t.get("plain_text", "") for t in rich_text_content])
+
+                # Try to parse as JSON for Data field, otherwise keep as string
+                if prop_name == "Data" and text:
+                    try:
+                        result[prop_name] = json.loads(text)
+                    except json.JSONDecodeError:
+                        result[prop_name] = text
```

Comment:
Can add a log here as why the json decoding failed?
Unless this is intentional and we're expecting this.